snuba/datasets/errors_replacer.py

```diff
@@ -662,16 +667,103 @@ def process_unmerge(
         "previous_group_id": message["previous_group_id"],
         "project_id": message["project_id"],
         "timestamp": timestamp.strftime(DATETIME_FORMAT),
+        "hashes": ", ".join(_convert_hash(h, state_name) for h in hashes),
     }
 
+    query_time_flags = (NEEDS_FINAL, message["project_id"])
+
+    return LegacyReplacement(
+        count_query_template, insert_query_template, query_args, query_time_flags
+    )
+
+
+def _convert_hash(
+    hash: str, state_name: ReplacerState, convert_types: bool = False
+) -> str:
     if state_name == ReplacerState.ERRORS:
-        query_args["hashes"] = ", ".join(
-            ["'%s'" % str(uuid.UUID(_hashify(h))) for h in hashes]
-        )
+        if convert_types:
+            return "toUUID('%s')" % str(uuid.UUID(_hashify(hash)))
+        else:
+            return "'%s'" % str(uuid.UUID(_hashify(hash)))
     else:
-        query_args["hashes"] = ", ".join("'%s'" % _hashify(h) for h in hashes)
+        if convert_types:
+            return "toFixedString('%s', 32)" % _hashify(hash)
+        else:
+            return "'%s'" % _hashify(hash)
 
-    query_time_flags = (NEEDS_FINAL, message["project_id"])
+
+def process_unmerge_hierarchical(
+    message: Mapping[str, Any],
+    all_columns: Sequence[FlattenedColumn],
+    state_name: ReplacerState,
+) -> Optional[Replacement]:
+    all_column_names = [c.escaped for c in all_columns]
+    select_columns = map(
+        lambda i: i if i != "group_id" else str(message["new_group_id"]),
+        all_column_names,
+    )
+
+    try:
+        timestamp = datetime.strptime(
+            message["datetime"], settings.PAYLOAD_DATETIME_FORMAT
+        )
+
+        primary_hash = message["primary_hash"]
+        assert isinstance(primary_hash, str)
+
+        hierarchical_hash = message["hierarchical_hash"]
+        assert isinstance(hierarchical_hash, str)
+
+        uuid.UUID(primary_hash)
+        uuid.UUID(hierarchical_hash)
+    except Exception:
+        # TODO(markus): We're sacrificing consistency over uptime as long as
+        # this is in development. At some point this piece of code should be
+        # stable enough to remove this.
+        logger.exception("process_unmerge_hierarchical.failed")
```

Comment:
please log the stack trace: 
See how it works here 
https://github.com/getsentry/snuba/pull/1904/files#diff-b2c8608d2750db0b89fc9bcb325e59164c4ca251e3a4eb582b103b7f19f009c5L522-L523

Reply:
I am updating this for style consistency but fwiw `logger.exception` adds `exc_info=True` which will grab the exception from `sys.exc_info()`.

Reply:
Interesting, I did not know that. Then your version was fine.